frontend/src/app/graphql.module.ts

```diff
@@ -0,0 +1,24 @@
+import { APOLLO_OPTIONS, ApolloModule } from 'apollo-angular';
+import { HttpLink } from 'apollo-angular/http';
+import { NgModule } from '@angular/core';
+import { ApolloClientOptions, InMemoryCache } from '@apollo/client/core';
+
+const uri = 'http://localhost:8000/api/graphql'; // <-- add the URL of the GraphQL server here
```

Comment:
This URL is now hardcoded. I can move it to `environment.ts` and `environment.prod.ts`, but how does this work when we start deploying? What URL should I pick for the production backend?

I'm sure you have run into this issue before in other projects, so any advice would be helpful!

Reply:
In most cases, you use absolute URLs without a hostname (+ port), e.g. `'/api/graphql'`. 

If you're accessing the frontend development server on `https://localhost:4200`, this will resolve as `https://localhost:4200/api/graphql`. The development server includes a proxy configuration, which will redirect that to `https://localhost:8000/api/graphql`, i.e. the backend port.

On production, you don't have this kind of proxying, as the endpoints for the frontend client and the backend API will have the same origin (e.g. `https://myproject.org` and `https://myproject.org/api/graphql`).

The [example in the apollo-angular documentation](https://the-guild.dev/graphql/apollo-angular/docs/data/network#usage) uses a URL like this, did you try that?

Reply:
Good find! I missed this in the documentation.